### PR TITLE
Add tools for PowerShell perf analysis

### PR DESCRIPTION
**Pull request by lzybkr**
## PR Summary

These scripts and additional files are useful in analyzing PowerShell performance on Windows with either perfview or wpr/wpa.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed - Issue link:
- **Testing - New and feature**
    - [x] Not Applicable or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

**Comment by powercode** [comment]
I'm so happy!!

Maybe add some info regarding using builds with symbols (cross gened) to get usable stacks?

File: tools/performance/GC.wprp
Comment (Robo210) [inline tools/performance/GC.wprp]: That's... 1 GB of buffers? Seems a bit excessive.

File: tools/performance/GC.wprp
Comment (Robo210) [inline tools/performance/GC.wprp]: Without a System collector this profile won't be usable by itself (it won't have process, thread, or loader events) - the user will always have to start at least one other session alongside this in order to make sense of the data in WPA.

File: tools/performance/README.md
Comment (Robo210) [inline tools/performance/README.md]: These wprp files aren't WPA regions files, they are WPR profiles.

File: tools/performance/PowerShellPerf.wprp
Comment (Robo210) [inline tools/performance/PowerShellPerf.wprp]: Your other profiles capture a small subset of the events this one does, but use a much larger set of buffers.

File: tools/performance/PowerShellPerf.wprp
Comment (Robo210) [inline tools/performance/PowerShellPerf.wprp]: Despite what the outdated MSDN documentation says, the Name attribute has been optional since Windows 8.1, and should be considered deprecated.

There's probably no value targeting the WPR that ships with the Windows 8 ADK as opposed to the Windows 8.1 ADK. Newer versions of the ADK can be installed on older OS releases.

File: tools/performance/Invoke-PerfviewPS.ps1
Comment (adityapatwardhan) [inline tools/performance/Invoke-PerfviewPS.ps1]: Can we make this a parameter? So we can launch a different pwsh.exe version than the one we are using.

File: tools/performance/GC.xml
Comment (adityapatwardhan) [inline tools/performance/GC.xml]: Please add EOL

File: tools/performance/GC.wprp
Comment (lzybkr) [reply tools/performance/GC.wprp]: Indeed. I forget where I copied that profile from and it looks like it's not actually being used so I'll just delete the file.

File: tools/performance/GC.wprp
Comment (lzybkr) [reply tools/performance/GC.wprp]: Yep, I'm removing this profile.

File: tools/performance/README.md
Comment (lzybkr) [reply tools/performance/README.md]: Yep, I clearly wasn't thinking when I wrote that - I'm removing the files.

File: tools/performance/PowerShellPerf.wprp
Comment (lzybkr) [reply tools/performance/PowerShellPerf.wprp]: This is the profile we care about. I don't recall seeing dropped events, so maybe this is fine?

File: tools/performance/PowerShellPerf.wprp
Comment (lzybkr) [reply tools/performance/PowerShellPerf.wprp]: Thanks, I appreciate the review. I agree, folks profiling PowerShell probably aren't running Win8 so I'll just remove the Name attribute.

File: tools/performance/PowerShell.stacktags
Comment (adityapatwardhan) [inline tools/performance/PowerShell.stacktags]: Could you add a comment why sometimes the module name has `.ni` while sometimes it doesn't

File: tools/performance/PowerShellPerf.wpaProfile
Comment (adityapatwardhan) [inline tools/performance/PowerShellPerf.wpaProfile]: Please add EOL

**Review by adityapatwardhan — CHANGES_REQUESTED** [review]

File: tools/performance/Invoke-PerfviewPS.ps1
Comment (lzybkr) [reply tools/performance/Invoke-PerfviewPS.ps1]: No, it's expensive and not normally needed, but good to know about so I left it. I'll add a comment to explain.

File: tools/performance/Invoke-PerfviewPS.ps1
Comment (lzybkr) [reply tools/performance/Invoke-PerfviewPS.ps1]: Sure.

File: tools/performance/PowerShell.stacktags
Comment (lzybkr) [reply tools/performance/PowerShell.stacktags]: Uh, I'm not sure, and maybe not relevant to PowerShell Core. This file isn't really that refined at all, so the hope is folks will contribute and make it more useful,

Comment (lzybkr) [reply tools/performance/PowerShell.stacktags]: I remember now, comment added.

**Comment by lzybkr** [comment]
Sorry about amending instead of adding, mostly just renaming and reformatting, but all feedback should be addressed now.

**Comment by anmenaga** [comment]
FYI. 18 spelling errors in tools/performance/README.md are causing Travis CI build failure.

**Comment by lzybkr** [comment]
I see 0 spelling errors and have no plans to make the tool happy, so a maintainer can fix or else close the PR.

**Comment by powercode** [comment]
The tool fails on stuff like `etl`, `perfview` etc. Only bogus errors.

**Review by adityapatwardhan — APPROVED** [review]

**Comment by lzybkr** [comment]
@powercode - yes, I reviewed what it complained about, it also complained about `analyze` I think.

But my point - we have humans reviewing and didn't complain. The tool should not block PRs.

**Comment by adityapatwardhan** [comment]
Updated the spellings file. Once CI is complete, I will merge.

**Comment by adityapatwardhan** [comment]
@lzybkr Thanks for your contribution!!

**Pull request merged**
